sidekick_core/lib/src/commands/update_command.dart

```diff
@@ -38,47 +80,109 @@ class UpdateCommand extends Command {
             ) ??
             Version.none;
 
-    if (versionToInstall <= currentSidekickCliVersion) {
+    final Version coreVersionToInstall;
+    if (version != null) {
+      if (!version.isPreRelease &&
+          !availableSidekickCoreVersion.contains(version)) {
+        print(
+          "'$version' is not a valid/compatible sidekick_core version, "
+          "visit https://pub.dev/packages/sidekick_core/versions for more info.",
+        );
+        return;
+      }
+      coreVersionToInstall = version;
+    } else {
+      if (availableSidekickCoreVersion.isEmpty) {
+        print('No compatible sidekick_core version found, '
+            'visit https://pub.dev/packages/sidekick_core/versions for more info.');
+        return;
+      }
+      if (availableSidekickCoreVersion.length == 1) {
+        coreVersionToInstall = availableSidekickCoreVersion.first;
+      } else {
+        print(white('Which sidekick_core version do you want to install?'));
+        coreVersionToInstall = menu(
```

Comment:
Currently this can print duplicate options. I added some prints so it's more clear when this happens:
```
❯ nh sidekick update
futureDartSdkVersions: [2.17.0, 2.17.1, 2.17.3, 2.17.5, 2.17.6, 2.17.7, 2.18.0, 2.18.1, 2.18.2, 2.18.3, 2.18.4, 2.18.5, 2.18.6, 2.18.7, 2.19.0, 2.19.1, 2.19.2, 2.19.3, 2.19.4, 2.19.5, 2.19.6, 3.0.0, 3.0.1, 3.0.2]
futureDartSdkVersionWithLatestPatch: [2.17.7, 2.18.7, 2.19.6, 3.0.2]
mappedSidekickCoreVersions: [1.2.0, 1.2.0, 1.2.0, null]
sdksWithValidCoreMapping: [MapEntry(2.17.7: 1.2.0), MapEntry(2.18.7: 1.2.0), MapEntry(2.19.6: 1.2.0)]
available: [1.2.0, 1.2.0, 1.2.0]
Which sidekick_core version do you want to install?
  1) 1.2.0 (latest)
  2) 1.2.0 (latest)
  3) 1.2.0 (latest)
```

I would prefer being able to choose the Dart version in this case, for example like this:
```
Which versions do you want to install?
  1) 1.2.0 (latest) with Dart 2.17.7
  2) 1.2.0 (latest) with Dart 2.18.7
  3) 1.2.0 (latest) with Dart 2.19.6 (latest) // selected as default
```

I have added a branch with some modifications to fix this here: https://github.com/phntmxyz/sidekick/commit/a082a4e4c7f66c5232abc9a2885b96df0bd9ebf5